### PR TITLE
[ADF-4275] - Fix preselect user validation for User Id

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-4275

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts [hunk 1]
```diff
@@ -190,7 +190,7 @@ export class PeopleCloudComponent implements OnInit, OnChanges {
                 this.logService.error(error);
             }
             const isUserValid: Boolean = this.userExists(result);
-            return isUserValid ? new IdentityUserModel(result[0]) : user;
+            return isUserValid ? new IdentityUserModel(result[0] || result) : user;
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts]: how is it possible that can e an array on a user?

Comment (SilviuCPopa) [reply lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts]: We have an array of Users when we find a user by email or by username and the api return more than one user

Comment (SilviuCPopa) [reply lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts]: I updated the way of finding the users

File: lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts [hunk 2]
```diff
@@ -190,17 +190,17 @@ export class PeopleCloudComponent implements OnInit, OnChanges {
...
     }
 
     async searchUser(user: IdentityUserModel) {
         const key: string = Object.keys(user)[0];
         switch (key) {
-            case 'id': return this.identityUserService.findUserById(user[key]).toPromise();
-            case 'username': return this.identityUserService.findUserByUsername(user[key]).toPromise();
-            case 'email': return this.identityUserService.findUserByEmail(user[key]).toPromise();
+            case 'id': return await this.identityUserService.findUserById(user[key]).toPromise();
+            case 'username': return (await this.identityUserService.findUserByUsername(user[key]).toPromise())[0];
```
Comment (eromano) [inline lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts]: (await this.identityUserService.findUserByUsername(user[key]).toPromise())[0]; it get the first of the array right? what happens if the array is empty?

File: lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts [hunk 3]
```diff
@@ -190,24 +192,36 @@ export class PeopleCloudComponent implements OnInit, OnChanges {
                 result = [];
                 this.logService.error(error);
             }
-            const isUserValid: boolean = this.userExists(result);
-            return isUserValid ? { isValid: isUserValid, user: new IdentityUserModel(user) } : { isValid: isUserValid, user: user };
+            const isUserValid: Boolean = this.userExists(result);
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/task/start-task/components/people-cloud/people-cloud.component.ts]: can you change to boolean

**Pull request merged**